pom.xml

```diff
@@ -92,48 +138,80 @@
         <profile>
             <id>AllTests</id>
             <properties>
-                <testng.suite.file>${testng.suite.path}/AllTestCalculatorHW1.xml</testng.suite.file>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/AllTestCalculatorHW1.xml
+                </maven-surefire-plugin.suiteXmlFile>
             </properties>
         </profile>
 
         <profile>
             <id>SumSub</id>
-            <activation>
-                <activeByDefault>true</activeByDefault>
-            </activation>
             <properties>
-                <testng.suite.file>${testng.suite.path}/SumSubTest.xml</testng.suite.file>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/SumSubTest.xml
+                </maven-surefire-plugin.suiteXmlFile>
             </properties>
         </profile>
 
         <profile>
             <id>SinCos</id>
             <properties>
-                <testng.suite.file>${testng.suite.path}/SinCosTest.xml</testng.suite.file>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/SinCosTest.xml
+                </maven-surefire-plugin.suiteXmlFile>
             </properties>
         </profile>
 
         <profile>
             <id>MulDiv</id>
             <properties>
-                <testng.suite.file>${testng.suite.path}/MulDivTest.xml</testng.suite.file>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/MulDivTest.xml
+                </maven-surefire-plugin.suiteXmlFile>
             </properties>
         </profile>
 
         <profile>
             <id>HW2Tests</id>
             <properties>
-                <testng.suite.file>${testng.suite.path}/AllTestsSeleniumHW2.xml</testng.suite.file>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/AllTestsSeleniumHW2.xml
+                </maven-surefire-plugin.suiteXmlFile>
             </properties>
         </profile>
 
         <profile>
             <id>HW3Tests</id>
             <properties>
-                <testng.suite.file>${testng.suite.path}/AllTestSeleniumHW3.xml</testng.suite.file>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/AllTestSeleniumHW3.xml
+                </maven-surefire-plugin.suiteXmlFile>
             </properties>
         </profile>
 
+        <profile>
+            <id>AllTestsSeleniumHW4</id>
+            <properties>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/AllTestsSeleniumHW4.xml
+                </maven-surefire-plugin.suiteXmlFile>
+            </properties>
+        </profile>
```

Comment:
Here you're trying to define property `maven-surefire-plugin.suiteXmlFile` using that property inside. It doesn't work. Maven fails with `NullPointerException` on this step.
Before you had 2 properties: `testng.suite.file` and `testng.suite.path` and it worked.